### ISSUE TITLE
can't wrap function in function - unknown symbol "this"

### ISSUE THREAD

**Issue by skorfmann** [winglang/wing#5065]
### I tried this:

```
bring cloud;

let authenticatedMiddleware = (handler: inflight (cloud.ApiRequest): cloud.ApiResponse): inflight (cloud.ApiRequest): cloud.ApiResponse => {
  let middleware = inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
    return handler(req);
  };

  return middleware;
};

```

see [playground](https://www.winglang.io/play?code=YgByAGkAbgBnACAAYwBsAG8AdQBkADsACgAKAGwAZQB0ACAAYQB1AHQAaABlAG4AdABpAGMAYQB0AGUAZABNAGkAZABkAGwAZQB3AGEAcgBlACAAPQAgACgAaABhAG4AZABsAGUAcgA6ACAAaQBuAGYAbABpAGcAaAB0ACAAKABjAGwAbwB1AGQALgBBAHAAaQBSAGUAcQB1AGUAcwB0ACkAOgAgAGMAbABvAHUAZAAuAEEAcABpAFIAZQBzAHAAbwBuAHMAZQApADoAIABpAG4AZgBsAGkAZwBoAHQAIAAoAGMAbABvAHUAZAAuAEEAcABpAFIAZQBxAHUAZQBzAHQAKQA6ACAAYwBsAG8AdQBkAC4AQQBwAGkAUgBlAHMAcABvAG4AcwBlACAAPQA%2BACAAewAKACAAIABsAGUAdAAgAG0AaQBkAGQAbABlAHcAYQByAGUAIAA9ACAAaQBuAGYAbABpAGcAaAB0ACAAKAByAGUAcQA6ACAAYwBsAG8AdQBkAC4AQQBwAGkAUgBlAHEAdQBlAHMAdAApADoAIABjAGwAbwB1AGQALgBBAHAAaQBSAGUAcwBwAG8AbgBzAGUAIAA9AD4AIAB7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAGgAYQBuAGQAbABlAHIAKAByAGUAcQApADsACgAgACAAfQA7AAoACgAgACAAcgBlAHQAdQByAG4AIABtAGkAZABkAGwAZQB3AGEAcgBlADsACgB9ADsACgA%3D)

### This happened:

![Screenshot 2023-11-27 at 20 44 25](https://github.com/winglang/wing/assets/136789/a692e548-42e1-477a-8e7e-c8efd6d8297c)

```
Failed to compile.

error: Unknown symbol "this"
  --> wing/main.w:1:1
  |
1 | bring cloud;
  | ^ Unknown symbol "this"
```

### I expected this:

no error

### Anything else?

Perhaps related to this, but I'm not sure https://github.com/winglang/wing/pull/4993

### Wing Version

0.50.6

### Node.js Version

18.7

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by skorfmann** [comment]
Looks a bit similar to https://github.com/winglang/wing/issues/3717 but this might be a coincidence

**Comment by Chriscbr** [comment]
The example in the original issue compiles and works now, but it will fail at runtime if the code is provided as a class static method instead of a free-floating function:

```js
bring cloud;

class A {
  pub static authenticatedMiddleware(handler: inflight (cloud.ApiRequest): cloud.ApiResponse): inflight (cloud.ApiRequest): cloud.ApiResponse {
    let middleware = inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
      return handler(req);
    };
    
    return middleware;
  }
}

let api = new cloud.Api();
api.get("/hello", A.authenticatedMiddleware(inflight (req) => {
  return { status: 200 };
}));
```

**Comment by yoav-steinberg** [comment]
> However, the alternate case where you write the code as a static method (instead of a free-floating function) causes a runtime error...

Fixed in #6040, closing.
